brutils/titulo_eleitoral.py

```diff
@@ -0,0 +1,192 @@
+
+    Returns:
+        tit_sequence[list]: sequential number sliced from input_string
+        tit_unid_fed[list]: Unidade Federal sliced from input_string
+        tit_dig_verifiers[list]: DV sliced from input_string
+    """
+    tit_sequence = input_string[:8]
+    tit_unid_fed = input_string[-4:-2]
+    tit_dig_verifiers = input_string[-2:]
+
+    return tit_sequence, tit_unid_fed, tit_dig_verifiers
+
+
+def _verify_length(numero_titulo, tit_unid_fed):
+    """
+    verify length of numero_titulo
+    considers edge case with 13 digits for SP & MG
+    Args:
+        numero_titulo[str]: an example of titulo eleitoral brasileiro
+        tit_unid_fed[list]: Unidade Federal sliced from input_string
+
+    Returns:
+        lentgh_verified[bool]: boolean indicating whether length of numero_titulo
+        is verified, or not.
+
+    """
+    if len(numero_titulo) == 12:
+        lentgh_verified = True
+
+    # edge case: for SP & MG with 9 digit long 'numero sequencial'
+    elif len(numero_titulo) == 13 and tit_unid_fed in ["01", "02"]:
+        lentgh_verified = True
+    else:
+        lentgh_verified = False
+
+    return lentgh_verified
+
+
+def _calculate_dv1(tit_sequence):
+    """calculate dv1
+    Args:
+        tit_sequence[list]: sequential number sliced from input_string
+    Returns:
+        v1[list]: list containing the resulting operation to calculate v1
+    """
+    x1, x2, x3, x4, x5, x6, x7, x8 = range(2, 10)
+
+    v1 = [
+        (int(tit_sequence[0]) * x1)
+        + (int(tit_sequence[1]) * x2)
+        + (int(tit_sequence[2]) * x3)
+        + (int(tit_sequence[3]) * x4)
+        + (int(tit_sequence[4]) * x5)
+        + (int(tit_sequence[5]) * x6)
+        + (int(tit_sequence[6]) * x7)
+        + (int(tit_sequence[7]) * x8)
+    ]
+
+    return v1
+
+
+def _verify_dv1(v1, tit_unid_fed, tit_dig_verifiers):
+    """extracts and verify dv1
+
+    Args:
+        v1[list]:
+        tit_unid_fed[list]: Unidade Federal sliced from input_string
+        tit_dig_verifiers[list]: DV sliced from input_string
+
+    Returns:
+        dv1[int]: result from v1 mod 11 operation
+        verified_dv1[bool]: boolean indicating whether dv1
+        is verified, or not.
+    """
+    dv1 = v1[0] % 11
+    # edge case: dv1 mod 11 == 0 and UF is SP or MG
+    if dv1 == 0 and tit_unid_fed not in ["01", "02"]:
+        dv1 = 1
+
+    # edge case: dv1 == 10, declare as zero instead
+    if dv1 == 10:
+        dv1 = 0
+
+    # verify dv1
+    return dv1, int(tit_dig_verifiers[0]) == dv1
+
+
+def _calculate_dv2(tit_unid_fed, dv1):
+    """calculate dv2
+
+    Args:
+        tit_unid_fed[list]: Unidade Federal sliced from input_string
+        dv1[int]: result from v1 mod 11 operation
+
+    Returns:
+        dv2[int]: result from v2 mod 11 operation
+
+    """
+    x9, x10, x11 = 7, 8, 9
+    v2 = [
+        (int(tit_unid_fed[0]) * x9) + (int(tit_unid_fed[1]) * x10) + (dv1 * x11)
+    ]
```

Comment:
Não parece necessário usar uma lista aqui.

```suggestion
    v2 = (
        (int(tit_unid_fed[0]) * x9) + (int(tit_unid_fed[1]) * x10) + (dv1 * x11)
    )
```